odl/solvers/nonsmooth/proximal_operators.py

```diff
@@ -138,17 +142,27 @@ def convex_conj_prox_factory(sigma):
 
         Parameters
         ----------
-        sigma : positive float
-            Step size parameter
+        sigma : positive float or componentwise positive space element
+                or list of positive floats
+            Step size parameter. Can be non-float if `prox_factory`
+            supports that.
 
         Returns
         -------
         proximal : `Operator`
             The proximal operator of ``s * F^*`` where ``s`` is the step size
         """
-        sigma = float(sigma)
-        prox_other = sigma * prox_factory(1.0 / sigma) * (1.0 / sigma)
-        return IdentityOperator(prox_other.domain) - prox_other
+
+        # Get the underlying space. At the same time, check if the given
+        # prox_factory accepts stepsize objects of the type given by sigma.
+        space = prox_factory(sigma).domain
+
+        mult_inner = MultiplyOperator(1.0 / sigma, domain=space, range=space)
+        mult_outer = MultiplyOperator(sigma, domain=space, range=space)
+        result = IdentityOperator(space) \
```

Comment:
Overall we prefer to use parentheses instead of \, this is mostly since it makes the code easier to read.

Reply:
bump

Reply:
To be honest, I am not sure what @adler-j means. Is it the line break? Should I put the product in parentheses and break the line in between?

Reply:
Yeah, we typically use

```
(looooooooooooooooooooooooooooooooooooooong + 
 liiiiiiiiiiiiiiiiiiinnnnnnnnnnnnnnneeeeeeeeeeeeeeee)
```

over

```
(looooooooooooooooooooooooooooooooooooooong \
 liiiiiiiiiiiiiiiiiiinnnnnnnnnnnnnnneeeeeeeeeeeeeeee)
```